lib/multiUrlGenerator.ts

```diff
@@ -3,7 +3,7 @@ type Component = { [replaceableKey: string]: string };
 export const multiUrlGenerator = (
   fullPath: string,
   components: Component[]
-): string[] => {
+): string[] | Promise<string[]> => {
```

Comment:
Let's remove this. We want this function to only return an array of strings. Ideally, functions should never return more than one possible type.